engine/src/main/java/org/hibernate/search/query/engine/spi/EntityInfo.java

```diff
@@ -23,9 +22,11 @@
 
 	Object[] getProjection();
 
-	List<Integer> getIndexesOfThis();
+	Iterable<Integer> getIndexesOfThis();
```

Comment:
I've already removed the commit, so we're talking about theory here.

In this specific case, we don't need empty/size checks. All we need is the client to iterate over it, so that's good. I understand it depends on the specific situation, I just think that in this specific case we don't need any of that, and even the unmodifiable collection is an unneeded complexity.